spec/other/trampoline_spec.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+require "spec_helper"
+
+describe "bundler version trampolining" do
+  before do
+    # ENV["BUNDLE_DISABLE_POSTIT"] = nil
+    FileUtils.rm_rf(system_gem_path)
+    FileUtils.cp_r(base_system_gems, system_gem_path)
+  end
+
+  context "version guessing" do
+    shared_examples_for "guesses" do |version|
+      it "guesses the correct bundler version" do
+        bundle! "--version"
+        expect(out).to eq("Bundler version #{version}")
+
+        if bundled_app("Gemfile").file?
+          bundle! "exec ruby -e 'puts Bundler::VERSION'"
+          expect(out).to eq(version)
+        end
+      end
+    end
+
+    context "with a lockfile" do
+      before do
+        install_gemfile ""
+        lockfile lockfile.sub(Bundler::VERSION, "1.12.0")
+      end
+
+      include_examples "guesses", "1.12.0"
+    end
+
+    context "with BUNDLER_VERSION" do
+      before do
+        ENV["BUNDLER_VERSION"] = "1.12.0"
+      end
+
+      context "with a lockfile" do
+        before { install_gemfile "" }
+        include_examples "guesses", "1.12.0"
+      end
+
+      context "without a lockfile" do
+        include_examples "guesses", "1.12.0"
+      end
+    end
+
+    context "with no hints" do
+      include_examples "guesses", Bundler::VERSION
+    end
+
+    context "with a gemfile and no lockfile" do
+      before do
+        gemfile ""
+      end
+
+      include_examples "guesses", Bundler::VERSION
+    end
+  end
+
+  context "installing missing bundler versions", :realworld => true do
+    before do
+      ENV["BUNDLER_VERSION"] = "1.12.3"
+    end
+
+    it "guesses & installs the correct bundler version" do
+      expect(system_gem_path.join("gems", "bundler-1.12.3")).not_to exist
+      bundle! "--version"
+      expect(out).to eq("Bundler version 1.12.3")
+      expect(system_gem_path.join("gems", "bundler-1.12.3")).to exist
+    end
+  end
+
+  context "bundle update --bundler" do
+    before do
+      simulate_bundler_version("1.11.1") do
+        install_gemfile ""
+      end
+    end
+
+    it "updates to the specified version" do
+      # HACK: since no released bundler version actually supports this feature!
+      bundle "update --bundler=1.12.0", :expect_err => true
+      expect(out).to include("Unknown switches '--bundler=1.12.0'")
```

Comment:
I'd update it as soon as we have a released version that actually accepts the `--bundle` option